site/src/modules/templates/useWatchVersionLogs.ts

```diff
@@ -0,0 +1,42 @@
+import { watchBuildLogsByTemplateVersionId } from "api/api";
+import { ProvisionerJobLog, TemplateVersion } from "api/typesGenerated";
+import { useState, useEffect } from "react";
+
+export const useWatchVersionLogs = (
+  templateVersion: TemplateVersion | undefined,
+  options?: { onDone: () => Promise<unknown> },
+) => {
+  const [logs, setLogs] = useState<ProvisionerJobLog[] | undefined>();
```

Comment:
I should've been more specific, but my original issue was really more wondering why we need `undefined` to be a unique state from `[]`. We could reset the state by just saying `setLogs([])` and the `onMessage` handler wouldn't need to check if `logs` is defined or not, it could just be simplified to `setLogs((logs) => [...logs, log])`.

Reply:
I use the following:
- `undefined` - connecting
- `[]` - connected but empty

Wdyt? I use the same idea for fetching server data. `query.data && <Loader />` and so on.